uxarray/core/dataarray.py

```diff
@@ -334,9 +360,48 @@ def to_raster(self, ax: GeoAxes):
         _ensure_dimensions(self)
 
         if not isinstance(ax, GeoAxes):
-            raise ValueError("`ax` must be an instance of cartopy.mpl.geoaxes.GeoAxes")
-
-        return _nearest_neighbor_resample(self, ax)
+            raise TypeError("`ax` must be an instance of cartopy.mpl.geoaxes.GeoAxes")
+
+        if pixel_mapping is not None:
+            if isinstance(pixel_mapping, xr.DataArray):
+                pixel_ratio_input = pixel_ratio
+                pixel_ratio = pixel_mapping.attrs["pixel_ratio"]
+                if pixel_ratio_input != 1 and pixel_ratio_input != pixel_ratio:
+                    warn(
+                        "Pixel ratio mismatch: "
+                        f"{pixel_ratio_input} passed but {pixel_ratio} in pixel_mapping. "
+                        "Using the pixel_mapping attribute.",
+                        stacklevel=2,
+                    )
```

Comment:
Should we validate whether the other attributes match with the provied `ax`? 
- `ax_projection`
- `ax_xlim`
- `ax_ylim`
- `ax_width`
- `ax_height`

We could include this logic within `_nearest_neighbor_resample` or as a small utility function (as we might support other sampling methods in the future.